app/models/binary/explore/NgffExplorer.scala

```diff
@@ -25,36 +25,66 @@ class NgffExplorer extends RemoteLayerExplorer {
+        } yield layers
+      })
+      layers: List[(ZarrLayer, Vec3Double)] = layerLists.flatten
     } yield layers
 
-  private def layerFromNgffMultiscale(multiscale: NgffMultiscalesItem,
-                                      remotePath: Path,
-                                      credentials: Option[FileSystemCredentials]): Fox[(ZarrLayer, Vec3Double)] =
+  private def getNgffMultiScaleChannelCount(multiscale: NgffMultiscalesItem, remotePath: Path): Fox[Int] =
+    for {
+      firstDataset <- multiscale.datasets.headOption.toFox
+      magPath = remotePath.resolve(firstDataset.path)
+      zarrayPath = magPath.resolve(ZarrHeader.FILENAME_DOT_ZARRAY)
+      zarrHeader <- parseJsonFromPath[ZarrHeader](zarrayPath) ?~> s"failed to read zarr header at $zarrayPath"
+      axisOrder <- extractAxisOrder(multiscale.axes) ?~> "Could not extract XYZ axis order mapping. Does the data have x, y and z axes, stated in multiscales metadata?"
+      channelAxisIndex <- axisOrder.c.toFox
+    } yield zarrHeader.shape(channelAxisIndex)
+
+  private def layersFromNgffMultiscale(multiscale: NgffMultiscalesItem,
+                                       remotePath: Path,
+                                       credentials: Option[FileSystemCredentials],
+                                       channelCount: Int): Fox[List[(ZarrLayer, Vec3Double)]] =
     for {
       axisOrder <- extractAxisOrder(multiscale.axes) ?~> "Could not extract XYZ axis order mapping. Does the data have x, y and z axes, stated in multiscales metadata?"
       axisUnitFactors <- extractAxisUnitFactors(multiscale.axes, axisOrder) ?~> "Could not extract axis unit-to-nm factors"
       voxelSizeInAxisUnits <- extractVoxelSizeInAxisUnits(
         multiscale.datasets.map(_.coordinateTransformations),
         axisOrder) ?~> "Could not extract voxel size from scale transforms"
-      magsWithAttributes <- Fox.serialCombined(multiscale.datasets)(d =>
-        zarrMagFromNgffDataset(d, remotePath, voxelSizeInAxisUnits, axisOrder, credentials))
-      _ <- bool2Fox(magsWithAttributes.nonEmpty) ?~> "zero mags in layer"
-      elementClass <- elementClassFromMags(magsWithAttributes) ?~> "Could not extract element class from mags"
-      boundingBox = boundingBoxFromMags(magsWithAttributes)
+      voxelSizeNanometers = voxelSizeInAxisUnits * axisUnitFactors
       nameFromPath <- guessNameFromPath(remotePath)
       name = multiscale.name.getOrElse(nameFromPath)
-      voxelSizeNanometers = voxelSizeInAxisUnits * axisUnitFactors
-      layer: ZarrLayer = if (looksLikeSegmentationLayer(name, elementClass)) {
-        ZarrSegmentationLayer(name, boundingBox, elementClass, magsWithAttributes.map(_.mag), largestSegmentId = None)
-      } else ZarrDataLayer(name, Category.color, boundingBox, elementClass, magsWithAttributes.map(_.mag))
-    } yield (layer, voxelSizeNanometers)
+      layerTuples <- Fox.combined(
+        (0 until channelCount)
+          .map(channelIndex => {
```

Comment:
```suggestion
      layerTuples <- Fox.serialCombined(0 until channelCount) { channelIndex =>
```
serialCombined is a shortcut for this exact case (we don’t have multiple foxes before, but want to iterate over a sequence, and with each item we will get a new fox by the block)